velox/functions/sparksql/specialforms/SparkCastHooks.cpp

```diff
@@ -16,14 +16,58 @@
 
 #include "velox/functions/sparksql/specialforms/SparkCastHooks.h"
 #include "velox/functions/lib/string/StringImpl.h"
-#include "velox/type/TimestampConversion.h"
+#include "velox/type/tz/TimeZoneMap.h"
 
 namespace facebook::velox::functions::sparksql {
 
+SparkCastHooks::SparkCastHooks(const core::QueryConfig& config) : CastHooks() {
+  const auto sessionTzName = config.sessionTimezone();
+  if (config.adjustTimestampToTimezone() && !sessionTzName.empty()) {
```

Comment:
heads up that `adjustTimestampToTimezone()` was added to be compatible with Presto's legacy_timestamp mode, which basically means that if you don't have a time zone associated with a timestamp, it would assume it was in the local user time zone. 

I'm not sure if this is the same behavior in Spark.

Reply:
We always set `adjustTimestampToTimezone = true` in gluten, and it is consistent with Spark's behavior.

Reply:
Makes sense, thanks for clarifying.